server/src/useCases/report/productTestCases/handleGetProductDetailsRequest.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-console */
+
+module.exports = class ReportGetProductDetailsRequestHandler {
+  constructor(request, response) {
+    this.req = request;
+    this.res = response;
+    this.dbConnect = request.app.locals.reportCollection;
+  }
+
+  async getProductDetails(repository) {
+    const { limit, offset } = this.req.query;
+    const { id } = this.req.params;
+
+    repository.aggregateBBDetailsByProductId({ id }, async (err, result) => {
+      if (err) {
+        console.error(err);
+        this.res.status(500).send(`Failed to fetch detailed report summary. Details: \n\t${err}`);
+        return;
+      }
+
+      const aggregatedResult = result[0];
+
+      /*  reduce to one item per endpoint, where we always prefer
+      the ones that didn't pass over the ones that did  */
+      const aggregatedData = aggregatedResult.data.reduce((items, item) => {
+        const lastItemIndex = items.length - 1;
+        const lastItem = items[lastItemIndex];
+
+        if (lastItem === undefined || item.uri !== lastItem.uri) {
```

Comment:
I imagine the undefined check is for the first time we run?

I would prefer to modify the lastItem assignment then - = lastItemIndex < 0 ? null : items{lastItemIndex[

the current way probably works, but it's because of JS quirks - I wouldn't count it on being 100% portable